railties/lib/rails/test_unit/runner.rb

```diff
@@ -48,7 +57,17 @@ def run(argv = [])
         def load_tests(argv)
           patterns = extract_filters(argv)
           tests = list_tests(patterns)
-          tests.to_a.each { |path| require File.expand_path(path) }
+          tests.to_a.each do |path|
+            require File.expand_path(path)
+          rescue LoadError => exception
+            all_tests = list_tests([default_test_glob])
+            corrections = DidYouMean::SpellChecker.new(dictionary: all_tests).correct(path)
```

Comment:
@bensheldon indeed. We need to check `exception.path` here to ensure it's the same path we tried to require.